### PR TITLE
Fix empty tensor when select -1

### PR DESCRIPTION
**Pull request by li-plus**
This PR fixed issue #1764 by first converting the `start` index to non-negative value before calculating the `end` index. The conversion is: `start += int(start < 0) * dimSize`. This can avoid `end` being 0 when `start` is -1.

I also added a test case `SelectIntNegativeDimAndIndexStaticModule_basic` that did not pass in LINALG backend without this PR to prove the correctness of my implementation.

### CODE REVIEW

**Comment by li-plus** [comment]
@ramiro050 Would you review this PR? Thanks.

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 1]
```diff
@@ -1347,6 +1347,18 @@ OpFoldResult AtenIntScalarOp::fold(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// AtenIntBoolOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenIntBoolOp::fold(ArrayRef<Attribute> operands) {
```
Comment (ramiro050) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1354]: Can you add e2e tests for this op too?

Comment (li-plus) [reply lib/Dialect/Torch/IR/TorchOps.cpp:1354]: Done! Added three e2e tests for this op.

**Comment by li-plus** [comment]
If everything is ok, would you please merge it for me? And I'll close the related issue.

**Comment by ZihengJiang** [comment]
Merged. Thanks @li-plus and @ramiro050

**Pull request merged**